core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -578,7 +579,8 @@ class Partition(val topicPartition: TopicPartition,
       // Updating the assignment and ISR state is safe if the partition epoch is
       // larger or equal to the current partition epoch.
       updateAssignmentAndIsr(
-        assignment = partitionState.replicas.asScala.map(_.toInt),
+        assignment = replicas,
+        followers = replicas.filter(_ != localBrokerId),
```

Comment:
Why not push this computation down to `updateAssignmentAndIsr`?

Reply:
Yeah, I debated this. I think that setting `followers` to `replicas` here and doing the filtering in `updateAssignmentAndIsr` is semantically wrong. A viable alternative would be to pass the `replicas` and an `isLeader` flag to `updateAssignmentAndIsr`. That would allow `updateAssignmentAndIsr` to compute the followers from the replicas when leader. I am not sure if it is really better than the suggested approach. I don't feel strong either ways so I am happy to go with your preferred approach.

Reply:
On second thought, it seems that the `isLeader` flag option is more efficient because it allows to clear the remote replicas map for followers.